Source/NSTableView.m

```diff
@@ -6940,6 +6948,84 @@ - (NSView *) makeViewWithIdentifier: (NSUserInterfaceItemIdentifier)identifier o
   return view;
 }
 
+- (id) _prototypeCellViewFromTableColumn: (NSTableColumn *)tb
+{
+  NSArray *protoCellViews = [tb _prototypeCellViews];
+  id view = nil;
+  
+  // it seems there is always one prototype...
+  if ([protoCellViews count] > 0)
+    {
+      view = [protoCellViews objectAtIndex: 0];
+      view = [view copy]; // instantiate the prototype...
+    }
+
+  return view;
+}
+
+- (NSTableRowView *) rowViewAtRow: (NSInteger)row makeIfNecessary: (BOOL)flag
+{
+  NSTableRowView *rv = nil;
+
+  if (_viewBased == YES)
+    {
+      NSNumber *aRow = [NSNumber numberWithInteger: row];
+      
+      rv = [_rowViews objectForKey: aRow];
+      if (rv == nil)
+	{
+	  if (flag == YES)
+	    {
+	      if ([_delegate respondsToSelector: @selector(tableView:rowViewForRow:)])
+		{
+		  rv = [_delegate tableView: self rowViewForRow: row];
+		}
+	      if (rv == nil)
+		{
+		  rv = AUTORELEASE([[NSTableRowView alloc] init]);
+		}	      
+	    }
+	  
+	  [_rowViews setObject: rv
+			forKey: aRow];
+	}
+    }
+  
+  return rv;
+}
+
+- (NSView *) viewAtColumn: (NSInteger)column row: (NSInteger)row makeIfNecessary: (BOOL)flag
+{
+  NSTableColumn *tb = [_tableColumns objectAtIndex: column];
+  NSIndexPath *path = [NSIndexPath indexPathForItem: column
+					  inSection: row];
+  NSView *view = [self _renderedViewForPath: path];
+  NSRect drawingRect = [self frameOfCellAtColumn: column
+					     row: row];
+  
+  // If the view has been stored use it, if not
+  // then grab it.
+  if (view == nil
+      && flag == YES)
+    {
+      if ([_delegate respondsToSelector: @selector(tableView:viewForTableColumn:row:)])
+	{
+	  view = [_delegate tableView: self
+		   viewForTableColumn: tb
+				  row: row];
+	}
+      else
+	{
+	  view = [self _prototypeCellViewFromTableColumn: tb];
+	}
+    }
+
+  [view setFrame: drawingRect];
+  [self _setRenderedView: view forPath: path];
```

Comment:
You could move this line into the if statement.

Reply:
Done.